File: codeforces/039/H.rb
n = gets.to_i
1.upto(n - 1){|i| puts 1.upto(n - 1).map{|j| (i * j).to_s(n)} * ' '}
__END__
# 	When 	Who 	Problem 	Lang 	Verdict 	Time 	Memory
396522 	Apr 19, 2011 6:57:24 PM 	watashi 	39H - Multiplication Table 	Ruby 	Accepted 	90 ms 	3800 KB

File: codeforces/044/H.hs
import Data.Char

main = do getLine >>= print . (\x -> sum (gao x) - if ok x then 1 else 0)  . map digitToInt

ok [a] = True
ok (a:b:c) = abs (a - b) <= 1 && ok (b:c)

gao [a] = take 10 $ repeat 1
gao (a:b:c) = [d!!j + if odd (i + b) then d!!(j+1) else 0 | i <- [0 .. 9], j <- [div (i + b) 2]] where
	d = gao (b:c)

--# 	When 	Who 	Problem 	Lang 	Verdict 	Time 	Memory
--404520 	Apr 21, 2011 3:02:55 PM 	watashi 	44H - Phone Number 	Haskell 	Accepted 	30 ms 	2000 KB 
